frag-nav/src/main/java/com/ncapdevi/fragnav/FragNavController.kt

```diff
@@ -306,6 +314,33 @@ class FragNavController constructor(private val fragmentManger: FragmentManager,
         }
     }
 
+    /**
+     * Push a fragment onto the current stack by fragment tag.
+     *
+     * @param fragmentClass      The fragment class to push.
+     * @param fragmentArgs       Optional Bundle to pass to the fragment as arguments
+     * @param transactionOptions Transaction options to be displayed
+     */
+    @JvmOverloads
+    fun <T : Fragment> pushFragment(fragmentClass: Class<T>, fragmentArgs: Bundle?, transactionOptions: FragNavTransactionOptions? = defaultTransactionOptions) {
+        if (currentStackIndex != NO_TAB) {
+            val ft = createTransactionWithOptions(transactionOptions, false)
+
+            removeCurrentFragment(ft, shouldDetachAttachOnPushPop(), shouldRemoveAttachOnSwitch())
+
+            val fragmentTag = generateTag(fragmentClass.javaClass)
+            fragmentStacksTags[currentStackIndex].push(fragmentTag)
+            ft.add(containerId, fragmentClass, fragmentArgs, fragmentTag)
+            commitTransaction(ft, transactionOptions)
+
+            val fragment = requireNotNull(fragmentManger.findFragmentByTag(fragmentTag)) { "Unable to find fragment with tag $fragmentTag" }
```

Comment:
I guess this might be tricky. Hopefully calling `findFragmentByTag` right after committing the transaction works as expected.

Reply:
Only if it's `commitNow`